src/engine/controls/bpmcontrol.cpp

```diff
@@ -795,19 +796,30 @@ mixxx::audio::FramePos BpmControl::getBeatMatchPosition(
         return thisPosition;
     }
 
-    const auto otherPosition = pOtherEngineBuffer->getExactPlayPos();
+    const mixxx::audio::FramePos otherPosition = pOtherEngineBuffer->getExactPlayPos();
     const mixxx::audio::SampleRate thisSampleRate = m_pBeats->getSampleRate();
 
     // Seek our next beat to the other next beat near our beat.
     // This is the only thing we can do if the track has different BPM,
     // playing the next beat together.
 
     // First calculate the position in the other track where this next beat will be.
-    const double thisSecs2ToNextBeat = (thisNextBeatPosition - thisPosition) /
-            thisSampleRate / thisRateRatio;
-    const mixxx::audio::FramePos otherPositionOfThisNextBeat = otherPosition +
-            thisSecs2ToNextBeat * otherBeats->getSampleRate() *
-                    pOtherEngineBuffer->getRateRatio();
+    // This original version has double precision issues (we kept it for documentation):
```

Comment:
Instead of leaving the old code as comment, src/test/bpmcontrol_test.cpp should be extended by a testcase that detects the precision issue.